modules/apps/frontend-js/frontend-js-tooltip-support-web/src/main/resources/META-INF/resources/index.tsx

```diff
@@ -34,7 +38,7 @@ const ALIGN_POSITIONS = [
 	'bottom-left',
 	'left',
 	'right',
-];
+] as const;
```

Comment:
`as const`: ❤️

Reply:
off topic comment, but what's the point `const ... as const`?

Reply:
Oh, sorry for asking. 
I guess this is called ["const assertions](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-4.html#:~:text=The%20const%20assertion%20allowed%20TypeScript,to%20use%20TypeScript's%20enum%20construct.)?" So now I get the point. Sorry about the noise

Reply:
So is this "similar" to `Object.freeze`?

Reply:
Yep, const assertions.

"Similar" to `Object.freeze` in the sense that it marks objects/arrays as `readonly` _in the type-system sense_. But _not_ in the runtime sense; the compiler doesn't compile this to `Object.freeze()` or anything under the covers. You can still come along in a (non-TS) JS file and mutate the values at runtime. But at least in TS files, the compiler will error if it sees you trying to write to such a value at compile-time.

The real benefit of `as const`, though, is that it prevents TypeScript from inferring undesirably broad types. For example, given an array like `['foo', 'bar']`, TS will infer a type of `Array<string>`. But with `as const` it will infer type as a read-only _tuple_ (eg. containing exactly 2 elements, immutable, and additionally that the values will be _exactly_ `'foo'` and `'bar'`). So in many places, the narrower/stricter type is exactly what you want and allows TS to detect many potential errors (as well as making reads easier too, requiring less hoop-jumping, casting, and edge-case-handling).

Reply:
Thanks, reading your comment made me understand more things in less time than the article I linked and that I was having a look at.

Reply:
Russian Novels: They Work™!